File: sdk/neutron_sdk.py
"""
Neutron SDK - Python Authentication Library
============================================
Use this SDK to authenticate users in your Python application.

Usage:
    from neutron_sdk import NeutronAuth
    
    auth = NeutronAuth()
    result = auth.login("username", "password", hwid="optional-hwid")
    
    if result['success']:
        print(f"Welcome, {result['user']['username']}!")
    else:
        print(f"Error: {result['error']}")
"""

import requests
import hashlib
import uuid


class NeutronAuth:
    def __init__(self):
        # Pre-configured credentials for your application
        self.api_url = "{{API_URL}}"
        self.app_secret = "{{APP_SECRET}}"
        self.app_name = "{{APP_NAME}}"
        self.version = "{{VERSION}}"
    
    def _get_hwid(self):
        """Generate a hardware ID based on system information."""
        try:
            return str(uuid.getnode())
        except:
            return "unknown"
    
    def login(self, username: str, password: str, hwid: str = None) -> dict:
        """
        Authenticate a user with the Neutron API.
        
        Args:
            username: The user's key/username
            password: The user's password
            hwid: Hardware ID (auto-generated if not provided)
        
        Returns:
            dict with 'success', 'user' or 'error' keys
        """
        if hwid is None:
            hwid = self._get_hwid()
        
        payload = {
            "secret": self.app_secret,
            "username": username,
            "password": password,
            "hwid": hwid
        }
        
        try:
            response = requests.post(
                f"{self.api_url}/login",
                json=payload,
                timeout=10
            )
            return response.json()
        except requests.exceptions.RequestException as e:
            return {"success": False, "error": str(e)}
    
    def check_license(self, license_key: str) -> dict:
        """
        Check if a license key is valid.
        
        Args:
            license_key: The license key to check
        
        Returns:
            dict with 'success' and license details or 'error'
        """
        payload = {
            "secret": self.app_secret,
            "license": license_key
        }
        
        try:
            response = requests.post(
                f"{self.api_url}/check",
                json=payload,
                timeout=10
            )
            return response.json()
        except requests.exceptions.RequestException as e:
            return {"success": False, "error": str(e)}


# Example usage
if __name__ == "__main__":
    auth = NeutronAuth()
    
    # Login example
    result = auth.login("your-key", "your-password")
    
    if result.get("success"):
        print(f"✓ Login successful!")
        print(f"  Username: {result['user']['username']}")
        print(f"  Expiry: {result['user']['expiry']}")
    else:
        print(f"✗ Login failed: {result.get('error')}")

File: sdk/NeutronSDK.cs
/*
 * Neutron SDK - C# Authentication Library
 * =========================================
 * Use this SDK to authenticate users in your C# / .NET application.
 * 
 * Usage:
 *     var auth = new NeutronAuth();
 *     var result = await auth.LoginAsync("username", "password");
 *     
 *     if (result.Success)
 *         Console.WriteLine($"Welcome, {result.User.Username}!");
 *     else
 *         Console.WriteLine($"Error: {result.Error}");
 */

using System;
using System.Net.Http;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;
using System.Management;

namespace NeutronSDK
{
    public class NeutronAuth
    {
        // Pre-configured credentials for your application
        private readonly string ApiUrl = "{{API_URL}}";
        private readonly string AppSecret = "{{APP_SECRET}}";
        private readonly string AppName = "{{APP_NAME}}";
        private readonly string Version = "{{VERSION}}";
        
        private readonly HttpClient _client;

        public NeutronAuth()
        {
            _client = new HttpClient();
            _client.Timeout = TimeSpan.FromSeconds(10);
        }

        private string GetHWID()
        {
            try
            {
                using (var searcher = new ManagementObjectSearcher("SELECT ProcessorId FROM Win32_Processor"))
                {
                    foreach (ManagementObject obj in searcher.Get())
                    {
                        return obj["ProcessorId"]?.ToString() ?? "unknown";
                    }
                }
            }
            catch { }
            return Environment.MachineName;
        }

        public async Task<LoginResult> LoginAsync(string username, string password, string hwid = null)
        {
            hwid ??= GetHWID();
            
            var payload = new
            {
                secret = AppSecret,
                username = username,
                password = password,
                hwid = hwid
            };

            try
            {
                var json = JsonSerializer.Serialize(payload);
                var content = new StringContent(json, Encoding.UTF8, "application/json");
                
                var response = await _client.PostAsync($"{ApiUrl}/login", content);
                var responseBody = await response.Content.ReadAsStringAsync();
                
                return JsonSerializer.Deserialize<LoginResult>(responseBody);
            }
            catch (Exception ex)
            {
                return new LoginResult { Success = false, Error = ex.Message };
            }
        }

        public async Task<LicenseResult> CheckLicenseAsync(string licenseKey)
        {
            var payload = new
            {
                secret = AppSecret,
                license = licenseKey
            };

            try
            {
                var json = JsonSerializer.Serialize(payload);
                var content = new StringContent(json, Encoding.UTF8, "application/json");
                
                var response = await _client.PostAsync($"{ApiUrl}/check", content);
                var responseBody = await response.Content.ReadAsStringAsync();
                
                return JsonSerializer.Deserialize<LicenseResult>(responseBody);
            }
            catch (Exception ex)
            {
                return new LicenseResult { Success = false, Error = ex.Message };
            }
        }
    }

    public class LoginResult
    {
        public bool Success { get; set; }
        public UserInfo User { get; set; }
        public string Error { get; set; }
    }

    public class UserInfo
    {
        public string Username { get; set; }
        public string Expiry { get; set; }
        public string Hwid { get; set; }
    }

    public class LicenseResult
    {
        public bool Success { get; set; }
        public LicenseInfo License { get; set; }
        public string Error { get; set; }
    }

    public class LicenseInfo
    {
        public string Key { get; set; }
        public bool IsUsed { get; set; }
        public string CreatedAt { get; set; }
    }
}
